src/Util.hpp

```diff
@@ -230,10 +231,62 @@ bool matches_dir_prefix_or_file(nonstd::string_view dir_prefix_or_file,
 // On Windows: Backslashes are replaced with forward slashes.
 std::string normalize_absolute_path(nonstd::string_view path);
 
-// Parse a string into an integer.
-//
-// Throws Error on error.
+/// Parse a string into an integer.
+/// Integer type must be passed as template parameter.
+///
+/// @param errorString for better error messages pass a user readable string.
+/// @throw Error on error.
```

Comment:
Sure, I will change it. But what exactly is the convention to be used? From what I see it's mostly informal, but I wanted to add a sentence specifically to that parameter.
Rather few places seem to document such stuff.

One example might be:

    // Write data from a buffer to the compressed stream.
    //
    // Parameters:
    // - data: Data to write.
    // - count: Size of data to write.
    //
    // Throws Error on failure.
    virtual void write(const void* data, size_t count) = 0;

Or I just drop the @param and @throw and write it as a sentence.

Reply:
Yes, the example you quoted is the style I used when I began C++-ifying code. @totph then [wanted to use a less verbose doc string style](https://github.com/ccache/ccache/pull/483#discussion_r342804473) (with private follow up mail discussion) when he worked on the Context PRs. Perhaps it was a mistake, but I didn't object since I didn't feel very strongly about it, and that's the style I've used since then.

What I meant more specifically by not introducing another doc string convention was to not use Doxygen-style markup (`///`, `@param`, etc) since that would be yet another style.

Here's what a @totph-style doc string could look like in this case:

```c++
// Parse `string` into an integer. The integer type is given by the template
// parameter. `error_string` should be a user-readable string describing the
// expected type.
//
// Throws `Error` on error.
template<class IntType>
IntType
stoi(const std::string& string, const char* error_string = "number")
```